db - possible data loss

Implicit conversion creates possible data loss:

1>C:\Repos\Stream-O-Vision\Stream-O-Vision\SqliteHandler.cpp(102,34): warning C4244: 'return': conversion from 'sqlite3_int64' to 'int', possible loss of data
1>C:\Repos\Stream-O-Vision\Stream-O-Vision\SqliteHandler.cpp(162,34): warning C4244: 'return': conversion from 'sqlite3_int64' to 'int', possible loss of data

--- a/Stream-O-Vision/SqliteHandler.cpp
+++ b/Stream-O-Vision/SqliteHandler.cpp
@@ -56,7 +56,7 @@ std::vector<Station> SqliteHandler::GetStationMap() {
 	return this->m_stations;
 }
 
-BOOL SqliteHandler::GetPlaylist(int id, int index) {
+BOOL SqliteHandler::GetPlaylist(int64_t id, int index) {
 	std::string strId = std::to_string(id).c_str();
 	std::string strQuery = "SELECT * FROM playlists WHERE fk_stations_id='" + strId + "';";
 	const char* query = strQuery.c_str();
@@ -87,7 +87,7 @@ BOOL SqliteHandler::GetPlaylist(int id, int index) {
 	return true;
 }
 
-int SqliteHandler::AddStation(std::string StationId, std::string StationName) {
+int64_t SqliteHandler::AddStation(std::string StationId, std::string StationName) {
 	std::string query = "INSERT INTO stations (StationId, StationName) VALUES ('" + StationId + "','" + StationName + "');";
 	char* err;
 	
@@ -102,7 +102,7 @@ int SqliteHandler::AddStation(std::string StationId, std::string StationName) {
 	return sqlite3_last_insert_rowid(this->m_sqliteDb);
 }
 
-BOOL SqliteHandler::DeleteStation(int dbStationId) {
+BOOL SqliteHandler::DeleteStation(int64_t dbStationId) {
 	std::string query = "DELETE FROM stations WHERE id=" + std::to_string(dbStationId) + ";";
 	char* err;
 
@@ -117,7 +117,7 @@ BOOL SqliteHandler::DeleteStation(int dbStationId) {
 	return true;
 }
 
-BOOL SqliteHandler::DeletePlaylistByStationId(int dbStationId) {
+BOOL SqliteHandler::DeletePlaylistByStationId(int64_t dbStationId) {
 	std::string query = "DELETE FROM playlists WHERE fk_stations_id=" + std::to_string(dbStationId) + ";";
 	char* err;
 
@@ -132,7 +132,7 @@ BOOL SqliteHandler::DeletePlaylistByStationId(int dbStationId) {
 	return true;
 }
 
-BOOL SqliteHandler::DeletePlaylistItemById(int dbPlaylistId) {
+BOOL SqliteHandler::DeletePlaylistItemById(int64_t dbPlaylistId) {
 	std::string query = "DELETE FROM playlists WHERE id=" + std::to_string(dbPlaylistId) + ";";
 	char* err;
 
@@ -147,7 +147,7 @@ BOOL SqliteHandler::DeletePlaylistItemById(int dbPlaylistId) {
 	return true;
 }
 
-int SqliteHandler::AddPlaylistItem(std::string filename, std::string path, int dbStationId) {
+int64_t SqliteHandler::AddPlaylistItem(std::string filename, std::string path, int64_t dbStationId) {
 	std::string query = "INSERT INTO playlists (Filename, Path, fk_stations_id) VALUES ('" + filename + "','" + path + "',"+ std::to_string(dbStationId) +");";
 	char* err;
 

--- a/Stream-O-Vision/SqliteHandler.h
+++ b/Stream-O-Vision/SqliteHandler.h
@@ -17,13 +17,13 @@ class SqliteHandler
 
 		BOOL LoadAll();
 		std::vector<Station> GetStationMap();
-		BOOL GetPlaylist(int,int);
+		BOOL GetPlaylist(int64_t,int);
 
-		int AddStation(std::string, std::string);
-		BOOL DeleteStation(int);
-		BOOL DeletePlaylistByStationId(int);
-		BOOL DeletePlaylistItemById(int);
-		int AddPlaylistItem(std::string, std::string, int);
+		int64_t AddStation(std::string, std::string);
+		BOOL DeleteStation(int64_t);
+		BOOL DeletePlaylistByStationId(int64_t);
+		BOOL DeletePlaylistItemById(int64_t);
+		int64_t AddPlaylistItem(std::string, std::string, int64_t);
 
 		BOOL CreateTables();
 

--- a/Stream-O-Vision/Stream-O-VisionView.cpp
+++ b/Stream-O-Vision/Stream-O-VisionView.cpp
@@ -20,7 +20,7 @@
 #include "AddStationDialog.h"
 #include "BroadcastSettingsDlg.h"
 #include "vlcpp/vlc.hpp"
-
+#include "grpc++/grpc++.h"
 
 
 
@@ -171,7 +171,7 @@ void CStreamOVisionView::OnBnClickedAddstation()
 		Station newStation = Station(); 
 		newStation.StationId = newStationDlg.GetStationId();
 		newStation.StationName = newStationDlg.GetStationName();
-		int rowId = Database.AddStation(CStringToStdString(newStation.StationId), CStringToStdString(newStation.StationName));
+		int64_t rowId = Database.AddStation(CStringToStdString(newStation.StationId), CStringToStdString(newStation.StationName));
 		newStation.dbStationId = rowId;
 		Stations.push_back(newStation);
 		UpdateStations();
@@ -182,7 +182,7 @@ void CStreamOVisionView::OnBnClickedAddstation()
 void CStreamOVisionView::OnBnClickedDeletestation()
 {
 	if (StationList.GetCurSel() != LB_ERR) {
-		int dbId = Stations[StationList.GetCurSel()].dbStationId; 
+		int64_t dbId = Stations[StationList.GetCurSel()].dbStationId; 
 		Database.DeleteStation(dbId);
 		Database.DeletePlaylistByStationId(dbId);
 		Stations.erase(StationList.GetCurSel() + Stations.begin());
@@ -224,7 +224,7 @@ void CStreamOVisionView::OnBnClickedAddmedia()
 			StationList.SetCurSel(0);
 		}
 		int index = StationList.GetCurSel();
-		int dbPlaylistId = Database.AddPlaylistItem(CStringToStdString(browseDlg.GetFileName()), CStringToStdString(browseDlg.GetPathName()), Stations[index].dbStationId);
+		int64_t dbPlaylistId = Database.AddPlaylistItem(CStringToStdString(browseDlg.GetFileName()), CStringToStdString(browseDlg.GetPathName()), Stations[index].dbStationId);
 		Stations[index].Media.push_back(MediaItem(browseDlg.GetPathName(), browseDlg.GetFileName(), dbPlaylistId));
 		UpdatePlaylistContents();
 	}

--- a/Stream-O-Vision/pch.h
+++ b/Stream-O-Vision/pch.h
@@ -22,9 +22,9 @@ typedef long int ssize_t;   // Need this for libvlcpp bindings to work right for
 struct MediaItem {
 	CString Path; 
 	CString Filename;
-	int dbPlaylistId;
+	int64_t dbPlaylistId;
 
-	MediaItem(CString path, CString filename, int dbId) {
+	MediaItem(CString path, CString filename, int64_t dbId) {
 		Path = path; 
 		Filename = filename;
 		dbPlaylistId = dbId;
@@ -41,7 +41,7 @@ struct Station {
 	std::vector<MediaItem> Media; 
 	CString StationName; 
 	CString StationId; 
-	int dbStationId;
+	int64_t dbStationId;
 	unsigned int MediaCurrentIndex; 
 
 	Station() {

